images/airflow/2.9.2/python/mwaa/celery/task_monitor.py

```diff
@@ -273,15 +286,48 @@ def _cleanup_undead_process(process_id: int):
     )
 
 
+def _get_next_unprocessed_signal() -> (str, dict):
+    signal_search_start_timestamp = math.ceil((datetime.now(tz=tz.tzutc()) - SIGNAL_SEARCH_TIME_RANGE).timestamp())
+    signal_filenames = os.listdir(MWAA_SIGNALS_DIRECTORY) if os.path.exists(MWAA_SIGNALS_DIRECTORY) else []
+    sorted_filenames = sorted(signal_filenames)
+    for signal_filename in sorted_filenames:
+        # Signals file follow the naming convention "<createTimestamp>_<signalId>.json"
+        file_timestamp = signal_filename.split("_")[0]
+        # In case of issues, signals can arrive late and out of order. So, we scan all unprocessed signals in a search time range.
+        # Only a handful of signals are expected for each worker, so this repeated processing should be very light.
+        if file_timestamp.isdigit() and int(file_timestamp) > signal_search_start_timestamp:
+            signal_file_path = os.path.join(MWAA_SIGNALS_DIRECTORY, signal_filename)
+            with open(signal_file_path, "r") as file_data:
```

Comment:
What user is this process running as?  The files will be written as read-only to anything outside of the sidecar, correct?

Reply:
In the serviceEni container implementation, the permissions are being explicitly changed for each file created. The Airflow container is able to read and write to the created files.